### PR TITLE
docker: add a minimum dockerfile

### PR DESCRIPTION
**Pull request by MegaMaddin**
This adds a Dockerfile which can be used as a base image in other projects to
support one-line-scan in a docker container.

Signed-off-by: Martin Mazein <amazein@amazon.de>

*Issue #, if available:*

*Description of changes:*

Adding a minimal dockerfile for inclusion in other projects.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by nmanthey — COMMENTED** [review]
Can you please add an example invocation to the commit message as well, i.e. an invocation via docker build and docker run?

File: tools/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:20.04
+
+ARG INFER_VERSION=v1.0.0
+
+# build dependencies
+RUN apt-get update -y
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install cppcheck git curl xz-utils
+
+# install one-line-scan
+RUN git clone https://github.com/awslabs/one-line-scan.git /opt/one-line-scan
```
[the remark this answers is not in the document]
Comment (MegaMaddin) [reply tools/Dockerfile:10]: My intention is to use this as a base image for other containers (see example).

**Comment by MegaMaddin** [comment]
> Can you please add an example invocation to the commit message as well, i.e. an invocation via docker build and docker run?

Added image build to the commit message. My example usage [in another project, e.g. KTF] would be:

- building the base image
```
docker build -t one-line-scan $URL
```

- building the project specific image on top
```
docker build -t ktf-one-line-scan --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) --build-arg USER=$USER --file tools/docker/OnelineScanDockerfile .
```

- adding this as invocation to the build chain

```
n$ make onelinescan
scanning current working directory with one-line-scan
Run diff analysis with given base commit mainline (9b390ef4d4f042f66a1cefcedf84137b7c746678)
Spotted 4 findings for base, and 4 for current commit


All findings worth reporting as introduced:



Did not detect relevant findings
Full stderr:
stderr.log:Versions of used tools:
stderr.log:Infer version v1.0.0
stderr.log:Copyright 2009 - present Facebook. All Rights Reserved.
stderr.log:Cppcheck 1.90
stderr.log:Failed infer analysis (on mainline), received exit code: 1
stderr.log:Failed CppCheck analysis (on mainline, might mean defects have been spotted), received exit code: 1
stderr.log:Failed infer analysis (on local_oneline), received exit code: 1
stderr.log:Failed CppCheck analysis (on local_oneline, might mean defects have been spotted), received exit code: 1
stderr.log:/opt/one-line-scan/one-line-cr-bot.sh: cleanup_handler() invoked, exit status 1
```

**Comment by nmanthey** [comment]
Thanks for sharing the example. As a side note, the environment variables that can be used to control the analysis in one-line-cr-bot would have to be forwarded to the container, in case those should be used.

Let's use this as a start, and in case we need more, let's iterate (i.e. not use cppcheck from Ubuntu) as that version is older than 2.2 that is available on github right now.

**Review by nmanthey — APPROVED** [review]

**Pull request merged**